src/opendr/perception/object_detection_2d/nanodet/nanodet_learner.py

```diff
@@ -0,0 +1,524 @@
+        if checkpoint_after_iter is not None:
+            self.cfg.schedule.val_intervals = checkpoint_after_iter
+        if checkpoint_load_iter is not None:
+            self.cfg.schedule.resume = checkpoint_load_iter
+        if temp_path != '':
+            self.cfg.save_dir = temp_path
+
+        self.cfg.freeze()
+
+    def save(self, path=None, verbose=True):
+        """
+        Method for saving the current model and metadata in the path provided.
+        :param path: path to folder where model will be saved
+        :type path: str, optional
+        :param verbose: whether to print a success message or not, defaults to False
+        :type verbose: bool, optional
+        """
+        path = path if path is not None else self.cfg.save_dir
+        model = self.cfg.check_point_name
+        os.makedirs(path, exist_ok=True)
+
+        metadata = {"model_paths": [], "framework": "pytorch", "format": "pth",
+                    "has_data": False, "inference_params": {}, "optimized": False,
+                    "optimizer_info": {}, "classes": self.classes}
+
+        param_filepath = "nanodet_{}.pth".format(model)
+        metadata["model_paths"].append(param_filepath)
+
+        logger = self.logger if verbose else None
+        if self.task is None:
+            print("You do not have call a task yet, only the state of the loaded or initialized model will be saved")
+            save_model_state(os.path.join(path, metadata["model_paths"][0]), self.model, None, logger)
+        else:
+            self.task.save_current_model(os.path.join(path, metadata["model_paths"][0]), logger)
+
+        with open(os.path.join(path, "nanodet_{}.json".format(model)), 'w', encoding='utf-8') as f:
+            json.dump(metadata, f, ensure_ascii=False, indent=4)
+        if verbose:
+            print("Model metadata saved.")
+        return True
+
+    def load(self, path=None, verbose=True):
+        """
+        Loads the model from the path provided.
+        :param path: path of the directory where the model was saved
+        :type path: str, optional
+        :param verbose: whether to print a success message or not, defaults to False
+        :type verbose: bool, optional
+        """
+        path = path if path is not None else self.cfg.save_dir
+        model = self.cfg.check_point_name
+        if verbose:
+            print("Model name:", model, "-->", os.path.join(path, model + ".json"))
+        with open(os.path.join(path, "nanodet_{}.json".format(model))) as f:
+            metadata = json.load(f)
+
+        logger = Logger(-1, path, False) if verbose else None
+        ckpt = torch.load(os.path.join(path, metadata["model_paths"][0]), map_location=torch.device(self.device))
+        self.model = load_model_weight(self.model, ckpt, logger)
+        if verbose:
+            logger.log("Loaded model weight from {}".format(path))
+        pass
+
+    def download(self, path=None, mode="image", verbose=False,
```

Comment:
"image" is not a valid model
```suggestion
    def download(self, path=None, mode="images", verbose=False,
```

Reply:
Thank you for your comment.